app/queries/keywords_query.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class KeywordsQuery
+  attr_reader :current_user, :filter_params
+
+  def initialize(current_user, filter_params)
```

Comment:
```suggestion
  def initialize(keywords = Keyword.all, filter_params = {})
```

A query object should work with any `scope`. We often have some previous filtering done before calling the query object:
- Authorization (pundit policies) will apply security rules
- Controller might have specific contexts (e.g. "CommunityKeywordsController" will NOT filter by `current_user` while `MyKeywordsController` would surely do)
- ...

The query object must only focus on applying filters (often defined by the request as params).


In our case, you would call the query object from the controller:

```
def index
  keywords = KeywordQuery.new(authorized_resources(Keyword), search_params)

  paginated_keywords = pagy(keywords, pagination_params)

  #...
end
```